pennylane/devices/default_qubit.py

```diff
@@ -537,83 +529,84 @@ def expval(self, observable, shot_range=None, bin_size=None):
         # intercept other Hamiltonians
         # TODO: Ideally, this logic should not live in the Device, but be moved
         # to a component that can be re-used by devices as needed.
-        if observable.name in ("Hamiltonian", "SparseHamiltonian"):
-            assert self.shots is None, f"{observable.name} must be used with shots=None"
-
-            self.map_wires(observable.wires)
-            backprop_mode = (
-                not isinstance(self.state, np.ndarray)
-                or any(not isinstance(d, (float, np.ndarray)) for d in observable.data)
-            ) and observable.name == "Hamiltonian"
-
-            if backprop_mode:
-                # TODO[dwierichs]: This branch is not adapted to broadcasting yet
-                if self._ndim(self.state) == 2:
-                    raise NotImplementedError(
-                        "Expectation values of Hamiltonians for interface!=None are "
-                        "not supported together with parameter broadcasting yet"
-                    )
-                # We must compute the expectation value assuming that the Hamiltonian
-                # coefficients *and* the quantum states are tensor objects.
-
-                # Compute  <psi| H |psi> via sum_i coeff_i * <psi| PauliWord |psi> using a sparse
-                # representation of the Pauliword
-                res = qml.math.cast(qml.math.convert_like(0.0, observable.data), dtype=complex)
-                interface = qml.math.get_interface(self.state)
-
-                # Note: it is important that we use the Hamiltonian's data and not the coeffs
-                # attribute. This is because the .data attribute may be 'unwrapped' as required by
-                # the interfaces, whereas the .coeff attribute will always be the same input dtype
-                # that the user provided.
-                for op, coeff in zip(observable.ops, observable.data):
-
-                    # extract a scipy.sparse.coo_matrix representation of this Pauli word
-                    coo = qml.operation.Tensor(op).sparse_matrix(wires=self.wires, format="coo")
-                    Hmat = qml.math.cast(qml.math.convert_like(coo.data, self.state), self.C_DTYPE)
-
-                    product = (
-                        self._gather(self._conj(self.state), coo.row)
-                        * Hmat
-                        * self._gather(self.state, coo.col)
-                    )
-                    c = qml.math.convert_like(coeff, product)
-
-                    if interface == "tensorflow":
-                        c = qml.math.cast(c, "complex128")
-
-                    res = qml.math.convert_like(res, product) + qml.math.sum(c * product)
+        if observable.name not in ("Hamiltonian", "SparseHamiltonian"):
```

Comment:
Is this all styling changes? It might be worth having a separate PR with just these changes, so that the diff on this PR focuses in just on those things we really care about.